.devcontainer/docker-compose.yml

```
@@ -0,0 +1,87 @@
+#
+# ---------------------------------------------------------------------
+#
+# GLPI - Gestionnaire Libre de Parc Informatique
+#
+# http://glpi-project.org
+#
+# @copyright 2015-2023 Teclib' and contributors.
+# @copyright 2003-2014 by the INDEPNET Development Team.
+# @licence   https://www.gnu.org/licenses/gpl-3.0.html
+#
+# ---------------------------------------------------------------------
+#
+# LICENSE
+#
+# This file is part of GLPI.
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <https://www.gnu.org/licenses/>.
+#
+# ---------------------------------------------------------------------
+#
+
+version: '3.9'
+
+services:
+  php:
+    container_name: "glpi-php-fpm"
+    build:
+      context: conf/php
+    restart: unless-stopped
+    volumes:
+      - ..:/var/www/:rw
```

Maybe we should use 2 distinct volumes here: a volume for GLPI codesource that uses the `read-only` mode, and a second volume (a named volume) for var files that uses the `read-write` mode.

```suggestion
      - ..:/var/www:ro
      - glpi-var:/var/glpi
```

In `Dockerfile`, you can define environment variable that permit to change the default var directories locations.
```
ENV \
  GLPI_CONFIG_DIR=/var/glpi/config \
  GLPI_VAR_DIR=/var/glpi/files
```

---

no, this is not a user docker config, we are in a development mode, we want to be able to edit glpi source files from vscode

---

I guess vscode is executed on host side, not in container side, so updating a source file in vscode should not trigger a writing operation inside the container.

---

I'll test but I'm not confident about that

---

So, confirmed, vscode directly opens the php container as its base root folder, not against the host.
When the editor opens a project in devcontainer mode, it starts the containers first, then open tree structure on the service defined in `devcontainer/devcontainer.json` (in our case php).

Not to be cocky, but to extends the feature with github codespace, theres is no Host access at all, vscode connect the remote container/service hosted in github DC.